### DISCUSSION TITLE
ILI9331 on bluepill_f103c8

### DISCUSSION THREAD

**Discussion by banoz** [moononournation/Arduino_GFX#176] [Q&A] [answered]
Hi,

I'm trying to make HelloWorld example to run on round ILI9331 display which is connected by parallel interface to STM32F103C8 based BluePill board. The init is like this:
```
/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_DataBus *bus = new Arduino_STM32PAR8(PB5, PB6, PB4, PB3, GPIOA);

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_GFX *gfx = new Arduino_ILI9331(bus, PB7, 0 /* rotation */, false /* IPS */);
```
and it doesn't work ofc.

My question is can Arduino_ILI9331 work on parallel interface at all?

Thanks,
Vas

**Comment by banoz** [comment]
The display is this board [https://www.aliexpress.com/item/2251832634390438.html](url)

**Comment by moononournation** [comment]
@monte-monte any suggestion?

**Comment by banoz** [comment]
and I've checked the display is not broken by running [this](https://github.com/prenticedavid/MCUFRIEND_kbv/blob/master/examples/LCD_ID_readreg/LCD_ID_readreg.ino) and getting the correct ID from reg 0x00

**Comment by monte-monte** [comment]
@banoz I have exactly the same display, and it works with bluepill. How did you wire it up exactly?

**Reply by banoz** [reply]
```
//Port data |D7 |D6 |D5 |D4 |D3 |D2 |D1 |D0 |
//Pin stm32 |PA7|PA6|PA5|PA4|PA3|PA2|PA1|PA0|

//Control pins |RD |WR |RS |CS |RST|
//Pin stm32    |PB3|PB4|PB5|PB6|PB7|
```

**Reply by banoz** [reply]
Here's the entire piece of code of the test that I'm running
```
#include <Arduino_GFX_Library.h>

#define GFX_BL DF_GFX_BL // default backlight pin, you may replace DF_GFX_BL to actual backlight pin

/* More dev device declaration: https://github.com/moononournation/Arduino_GFX/wiki/Dev-Device-Declaration */
#if defined(DISPLAY_DEV_KIT)
Arduino_GFX *gfx = create_default_Arduino_GFX();
#else /* !defined(DISPLAY_DEV_KIT) */

/* More data bus class: https://github.com/moononournation/Arduino_GFX/wiki/Data-Bus-Class */
Arduino_DataBus *bus = new Arduino_STM32PAR8(PB5, PB6, PB4, PB3, GPIOA);

/* More display class: https://github.com/moononournation/Arduino_GFX/wiki/Display-Class */
Arduino_GFX *gfx = new Arduino_ILI9331(bus, PB7, 0 /* rotation */, false /* IPS */);

#endif /* !defined(DISPLAY_DEV_KIT) */
/*******************************************************************************
 * End of Arduino_GFX setting
 ******************************************************************************/

void setup(void)
{
    gfx->begin();
    gfx->fillScreen(BLACK);

#ifdef GFX_BL
    pinMode(GFX_BL, OUTPUT);
    digitalWrite(GFX_BL, HIGH);
#endif

    gfx->setCursor(10, 10);
    gfx->setTextColor(RED);
    gfx->println("Hello World!");

    pinMode(PC13, OUTPUT);

    delay(5000); // 5 seconds
}

void loop()
{
    gfx->setCursor(random(gfx->width()), random(gfx->height()));
    gfx->setTextColor(random(0xffff), random(0xffff));
    gfx->setTextSize(random(6) /* x scale */, random(6) /* y scale */, random(2) /* pixel_margin */);
    gfx->println("Hello World!");

    delay(1000); // 1 second
    digitalWrite(PC13, !digitalRead(PC13)); // toggle PC13
}
```

**Reply by banoz** [reply]
Just tried to change gfx->fillScreen(BLACK) to gfx->fillScreen(YELLOW) and it worked. But still no text on the screen.

**Comment by monte-monte** [comment]
@moononournation I've tested on my setup, that was laying around from the time I've made a commit https://github.com/moononournation/Arduino_GFX/pull/112 and it turns out that it doesn't work with version 1.1.9 that you've released after the one I was working with. So I guess something you've added apart from my commits broke the part I've wrote.
To be clear, I've just tested version 1.1.9 - not working, just after first text appears, everything stops and display is cleared, seems like MCU just crashes.
Then I've downloaded 1.1.8 and added only my commits for ILI9331 and STM32 parallel support. And now everything works again.

**Comment by monte-monte** [comment, marked as answer]
@moononournation https://github.com/moononournation/Arduino_GFX/commit/aa8621f6142cfd05ddc86ab28464d58c56cdb2e4 this commit brakes things.

**Reply by monte-monte** [reply]
@banoz you can try pull this and see if it will work for you https://github.com/moononournation/Arduino_GFX/pull/177

**Reply by banoz** [reply]
It helped, thank you!

**Reply by banoz** [reply]
Found a copy-paste issue 
`#define ILI9331_CABCCTL7 0xCD     ///CABC Control 7`
should be 
`#define ILI9331_CABCCTL7 0xCE     ///CABC Control 7`

**Comment by moononournation** [comment]
I just found there are double checking on the changes of the address window, please get update if it can fit it.
Sorry for I do not have ILI9331 and Bluepill in hand.

**Reply by monte-monte** [reply]
Yes the latest changes to ILI9331 works. Rotation is fine too.
